base_user_role/views/user.xml

```diff
@@ -39,4 +39,14 @@
             </field>
         </field>
     </record>
+    <record id="view_res_users_tree_inherit" model="ir.ui.view">
+        <field name="name">res.users.tree.inherit</field>
+        <field name="model">res.users</field>
+        <field name="inherit_id" ref="base.view_users_tree" />
+        <field name="arch" type="xml">
+            <field name="company_id" position="after">
```

Comment:
```suggestion
            <field name="company_id" position="before">
```

In odoo, the company_id field is usually the last column of the tree view. So better to keep that design, and set the value before.